### PR TITLE
Open oauth callback endpoints to be public

### PR DESCRIPTION
**Pull request by krynble**
Community issue:
https://community.n8n.io/t/oauth-authorization-error-zendesk-oauth2-api-credential/12439

### CODE REVIEW

File: packages/cli/src/UserManagement/routes/index.ts [hunk 1]
```diff
@@ -66,6 +66,8 @@ export function addRoutes(this: N8nApp, ignoredEndpoints: string[], restEndpoint
 			req.url.startsWith(`/${restEndpoint}/forgot-password`) ||
 			req.url.startsWith(`/${restEndpoint}/resolve-password-token`) ||
 			req.url.startsWith(`/${restEndpoint}/change-password`) ||
+			req.url.startsWith(`/${restEndpoint}/oauth2-credential/callback`) ||
+			req.url.startsWith(`/${restEndpoint}/oauth1-credential/callback`) ||
```
Comment (hasan-dot) [inline packages/cli/src/UserManagement/routes/index.ts:70]: Makes total sense to add these as a default non authenticated routes.

File: packages/cli/src/requests.d.ts [hunk 2]
```diff
@@ -237,12 +237,16 @@ export declare namespace OAuthRequest {
...
+			user?: User;
+		};
 	}
 
 	namespace OAuth2Credential {
 		type Auth = OAuth1Credential.Auth;
-		type Callback = AuthenticatedRequest<{}, {}, {}, { code: string; state: string }>;
+		type Callback = AuthenticatedRequest<{}, {}, {}, { code: string; state: string }> & {
+			user?: User;
+		};
```
Comment (hasan-dot) [inline packages/cli/src/requests.d.ts:249]: This is exactly what I was searching for, but couldn't find it

**Review by hasan-dot — COMMENTED** [review]
Always glad to learn something new! Thanks for the PR

**Comment by janober** [comment]
Thanks a lot. Got merged.

**Comment by janober** [comment]
Got released with `n8n@0.174.0`

**Pull request merged**